pandas/core/resample.py

```diff
@@ -1519,9 +1519,27 @@ def _get_period_bins(self, ax):
 
         start = ax.min().asfreq(self.freq, how=self.convention)
         end = ax.max().asfreq(self.freq, how='end')
+        if self.base:
+            start = start.to_timestamp()
+            end = end.to_timestamp()
+
+            # get base adjusted bin edge labels
```

Comment:
I would move the PI creation into the if & the else blocks (as the else is almost a trivial creation then).

Reply:
agreed and done, despite the duplication of the PI creation I think it makes it just that much more obvious/readable.